### PR TITLE
[SPARK-49457][BUILD] Remove uncommon curl option `--retry-all-errors`

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Remove uncommon curl option `--retry-all-errors`, which is added in curl 7.71.0 - June 24 2020, old versions can not recognize this option.

### Why are the changes needed?

It causes `build/mvn` to fail on Ubuntu 20.04.

```
exec: curl --retry 3 --retry-all-errors --silent --show-error -L https://www.apache.org/dyn/closer.lua/maven/maven-3/3.9.9/binaries/apache-maven-3.9.9-bin.tar.gz?action=download
curl: option --retry-all-errors: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
```

```
$ curl --version
curl 7.68.0 (aarch64-unknown-linux-gnu) libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3
Release-Date: 2020-01-08
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS brotli GSS-API HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually test.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by pan3793** [comment]
cc @yaooqinn @LuciferYang

File: dev/create-release/release-build.sh [hunk 1]
```diff
@@ -536,7 +536,7 @@ if [[ "$1" == "publish-release" ]]; then
       file_short=$(echo $file | sed -e "s/\.\///")
       dest_url="$nexus_upload/org/apache/spark/$file_short"
       echo "  Uploading $file_short"
-      curl --retry 3 --retry-all-errors -u $ASF_USERNAME:$ASF_PASSWORD --upload-file $file_short $dest_url
+      curl --retry 3 -u $ASF_USERNAME:$ASF_PASSWORD --upload-file $file_short $dest_url
```
[the remark this answers is not in the document]
Comment (pan3793) [reply dev/create-release/release-build.sh:539]: the release script should always run inside the docker container, so we don't need to make it support broad versions, how about reserving this only removing `--retry-all-errors` from `build/mvn`?

[the remark this answers is not in the document]
Comment (LuciferYang) [reply dev/create-release/release-build.sh:539]: fine to me

[the remark this answers is not in the document]
Comment (pan3793) [reply dev/create-release/release-build.sh]: okay, I reverted the change in this file, thanks for reviewing.

**Review by LuciferYang — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master. Thanks @pan3793 and @HyukjinKwon

**Pull request closed without merging**